cmd/create/machinepool/nodepool.go

```diff
@@ -121,6 +122,45 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		npBuilder = npBuilder.Replicas(replicas)
 	}
 
+	// Machine pool instance type:
+	// NodePools don't support MultiAZ yet, so the availabilityZonesFilters is calculated from the cluster
+	availabilityZonesFilter := cluster.Nodes().AvailabilityZones()
+	instanceType := args.instanceType
+	instanceTypeList, err := r.OCMClient.GetAvailableMachineTypesInRegion(cluster.Region().ID(),
+		availabilityZonesFilter, cluster.AWS().STS().RoleARN(), r.AWSClient)
+	if err != nil {
+		r.Reporter.Errorf(fmt.Sprintf("%s", err))
+		os.Exit(1)
+	}
+
+	if interactive.Enabled() {
```

Comment:
should we enter interactive mode if the instance type is not specified?

Reply:
According to the "classic machinepool" logic, no.

The "nodepool.go" has some duplication of the logic due to convenience, but the behavior would be similar (or pretty similar on common features, as "hosted machinepools" have specific features).

Once we consolidate the pending features (MultiAZ, subnet, preflight), we can extract the common flows in a future iteration.